### DISCUSSION TITLE
Windows testing and resolving outstanding issues

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#668] [General]
Hi Windows users,

We have a Windows GLSLang related build issue and vsgmultiviews issue reported on VulkanSceneGraph and vsgExamples Issue trackers.  I have done as much as I can to tease out info from engineers posting the issues but can't go any further as I don't have a Windows system for testing, nor expertise on Windows specific issues.  At this point I don't know what the issues are caused by, so there is no way for me to try and address them, so really need help getting to the bottom of these issues.

The two issues are:

- [ShaderCompiler.cpp(197,5): error C2440: 'initializing': cannot convert from 'initializer list' to 'int'](https://github.com/vsg-dev/VulkanSceneGraph/issues/658)
- [vsgmultiviews is not ok](https://github.com/vsg-dev/vsgExamples/issues/196)

Thanks in advance for you help,
Robert.

**Comment by robertosfield** [comment]
Another item it would be useful to get feedback on is a modification that I made to @rhabacker's original mod for MinGW DLL versioning so that it's used for all Win32 builds, the respective PR's are:

- [Changed the vsg--${VSG_SOVERSION} dll name to be applied to all Windows builds.](https://github.com/vsg-dev/VulkanSceneGraph/pull/664)
- [Create and install shared library with version on MinGW](https://github.com/vsg-dev/VulkanSceneGraph/pull/663) 

This change changes the vsg.dll to vsg-SOVERSION.dll with the associated vsg.lib (or vsg.dll.lib for MinGW) linking to this versions .dll.  This goes a little way towards the versioning that all other platform enjoy.

Hopefully all those building shader libs under Windows won't see any issues, things will just work as before but hopefully a little more reliably.  I won't assume everything is golden though, community awareness and testing of these changes now in VSG master would be much appreciated.

If things with VSG master look solid I'll merge with the VulkanSceneGraph-1.0 branch and then look at 1.0.3 release before we all head offline to join our family and friends for the festive season.

**Comment by robertosfield** [comment]
I would like to get to the bottom of the reported Windows build error #658, for some reason it looks like CMake's tests of GLSLang isn't resolve the GLSLang features correctly.

Could Windows users please post the info about the CMake/VisualStudio/VulkanSDK versions they are using, and the CMakeCache.txt file that is generated when they run, and whether it builds fine or not.

I'm hoping we might be able to spot a pattern to when and where things work vs fail and from there figure out a tweak to CMake to address the problem, or add some extra install documentation to help avoid errors when setting things up.

**Comment by 640kb** [comment]
When I try to compile vsgFramework with -DBUILD_SHARED_LIBS=ON, I get the following error.

```
[ 97%] ←[32m←[1mLinking CXX executable ..\bin\unit.exe←[0m
[ 97%] Built target unit
[ 97%] ←[34m←[1mGenerating D:/projects/git/vsgFramework/build-debug/include/glslang/glsl_intrinsic_header.h←[0m
Error: dependent 'components\glslang-build\SPIRV\SPIRVd.lib' does not exist.
jom: D:\projects\git\vsgFramework\build-debug\CMakeFiles\Makefile2 [components\glslang-build\StandAlone\CMakeFiles\glslangValidator.dir\all] Error 2
jom: D:\projects\git\vsgFramework\build-debug\Makefile [all] Error 2
```

This is because SPIRV cannot be compiled correctly as a DLL yet. There is also a ticket in the tracker of the glslang project.
https://github.com/KhronosGroup/glslang/issues/2283
